keg_bouncer/db/__init__.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import absolute_import
+
+import sqlalchemy as sa
+import sqlalchemy.orm as saorm
+from keg.db import db
+from keg_elements.db.mixins import MethodsMixin
+
+
+class Permission(db.Model, MethodsMixin):
+    """An entity that describes an authentication boundary which can be used directly
+    in the application to permit or deny access to something.
+
+    As a whole, the set of Permission entities describes every possible authentication
+    boundary within the application.
+
+    An "authentication boundary" is a decision point where the question arises,
+    "Is the current user of this application authorized to access this feature?"
+
+    Permissions have two touchpoints with the application:
+      1) They touch the source code directly.
+      2) They touch the user of the application directly.
+
+    Source code includes decision points where a particular permission can be required
+    for a user to access something. To facilitate this touchpoint, each permission has
+    a unique `token` string which allows the code to refer to the permission itself.
+    This is the only way, in fact, that source code can refer to a permission.
+
+    Permissions also directly touch users of the application in that users can be
+    granted any number of permissions. These represent the authority possesed by the
+    user. To facilitate this touchpoint, each permission has a string label that
+    provides a helpful, human-oriented description of the permission.
+    """
+    __tablename__ = 'permissions'
+    id = sa.Column(sa.Integer, primary_key=True)
+    token = sa.Column(sa.String(), nullable=False, unique=True)
+    label = sa.Column(sa.Unicode(), nullable=False)
```

Comment:
> optional, a length for the column for use in DDL and CAST expressions. 

This will be DB specific. In Postgres that means a string of unlimited size.